agent/agentcontainers/dcspec/gen.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# This script requires quicktype to be installed.
+if ! command -v quicktype &>/dev/null; then
+	echo "quicktype could not be found. Please install it to use this script:"
+	echo ">>> npm install -g quicktype"
+	exit 1
+fi
+
+DEST_FILENAME="dcspec_gen.go"
+SCRIPT_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+DEST_PATH="${SCRIPT_DIR}/${DEST_FILENAME}"
+SCHEMA_SRC="https://raw.githubusercontent.com/devcontainers/spec/refs/heads/main/schemas/devContainer.base.schema.json"
+
+# Location of the JSON schema for the devcontainer specification.
+TMPDIR=$(mktemp -d)
+trap 'rm -rfv "$TMPDIR"' EXIT
+curl --fail --silent --show-error --location --output "${TMPDIR}/schema.json" "${SCHEMA_SRC}"
+rm -f "${DEST_PATH}"
+quicktype \
+	--src-lang schema \
+	--lang go \
+	--just-types-and-package \
+	--top-level "DevContainer" \
+	--out "${DEST_PATH}" \
+	--package "dcspec" \
+	"${TMPDIR}/schema.json"
```

Comment:
Could we follow this up with `gofumpt`? It might be painful to work with this if editing it results in a full re-format/indentation of the file.